pkg/cache/v3/snapshot.go

```diff
@@ -66,15 +66,15 @@ func IndexRawResourcesByName(items []types.Resource) map[string]types.Resource {
 
 // NewResources creates a new resource group.
 func NewResources(version string, items []types.Resource) Resources {
-	itemsWithTtl := []types.ResourceWithTtl{}
+	itemsWithTTL := []types.ResourceWithTtl{}
 	for _, item := range items {
-		itemsWithTtl = append(itemsWithTtl, types.ResourceWithTtl{Resource: item})
+		itemsWithTTL = append(itemsWithTTL, types.ResourceWithTtl{Resource: item})
 	}
-	return NewResourcesWithTtl(version, itemsWithTtl)
+	return NewResourcesWithTtl(version, itemsWithTTL)
 }
 
 // NewResources creates a new resource group.
-func NewResourcesWithTtl(version string, items []types.ResourceWithTtl) Resources {
+func NewResourcesWithTtl(version string, items []types.ResourceWithTtl) Resources { // nolint:golint,revive
```

Comment:
Consistency with the `TTL` casing

Reply:
So Go naming conventions really want this to be `TTL`, but I didn't want to break any compatibility in this PR. Since this is a public function, I just annotated it.

Maybe rename and deprecate?

```Go
// Deprecated 
func NewResourcesWithTtl(version string, items []types.ResourceWithTtl) Resources {
        return NewResourcesWithTTL(...)
}

func NewResourcesWithTTL(version string, items []types.ResourceWithTtl) Resources {
...
}
```

Reply:
I'm all for the rename. Lets make it consistent throughout

Reply:
AFAIK we can't add compatibility aliases since we need to rename the `Ttl` field in `ResourceWithTtl` and there's no way to alias that.